hw/chisel/doc/xdma.md

```diff
@@ -0,0 +1,238 @@
+# Snax Cluster eXtendable Distributed Memory Access (XDMA)
+
+This documentation introduces the Extendable Distributed Memory Access (XDMA), a system designed for widespread use in Snax clusters, facilitating both intra-cluster and inter-cluster *Computation on Copy (CoC)*. The documentation covers the following aspects:
+
+- [Snax Cluster eXtendable Distributed Memory Access (XDMA)](#snax-cluster-extendable-distributed-memory-access-xdma)
+  - [Introduction and XDMA hardware block diagram](#introduction-and-xdma-hardware-block-diagram)
+  - [Source Code Hierarchy](#source-code-hierarchy)
+  - [Features](#features)
+    - [New Data-fetching Unit (Streamer) to Support Powerful Direct Memory Access](#new-data-fetching-unit-streamer-to-support-powerful-direct-memory-access)
+    - [Extendable Datapath Extension to Enable *Computation on Copy* Capability](#extendable-datapath-extension-to-enable-computation-on-copy-capability)
+    - [Distributed Datapath for Cluster-to-Cluster or Host-to-Cluster Memory Access](#distributed-datapath-for-cluster-to-cluster-or-host-to-cluster-memory-access)
+  - [CSR Address Definition](#csr-address-definition)
+  - [Design of Custom Datapath Extension](#design-of-custom-datapath-extension)
+    - [Native Datapath Extension](#native-datapath-extension)
+    - [SystemVerilog Datapath Extension](#systemverilog-datapath-extension)
+  - [Integration with SNAX Platform](#integration-with-snax-platform)
+    - [Hardware Interface](#hardware-interface)
+    - [Software Interface](#software-interface)
+
+
+## Introduction and XDMA hardware block diagram
+
+XDMA is implemented in every cluster with the following architecture:
+
+![XDMA Hardware Diagram](xdma_figures/xdma_datapath.png)
```

Comment:
So that you don't need to save pictures, you can copy-paste and paste them directly to the github page! So it goes like:

1. Take a screen shot of your image.
2. Paste it here directly on the .md in github.
3. Github creates its own link to the image.

Just a suggestion. I was told to do that better than to save pictures here all the time.

Reply:
yes this is better

Reply:
Haha okay! I wrote the markdown file offline (the preview renderring happens on my computer locally), so I don't know github natively supports it...